plugins/orchestrator/src/api/OrchestratorClient.ts

```diff
@@ -155,14 +121,49 @@ export class OrchestratorClient implements OrchestratorApi {
   }): Promise<WorkflowExecutionResponse> {
     const baseUrl = await this.getBaseUrl();
     const urlToFetch = `${baseUrl}/instances/${args.instanceId}/retrigger`;
-    const response = await fetch(urlToFetch, {
+    return await this.fetcher(urlToFetch, {
       method: 'POST',
       body: JSON.stringify(args.inputData),
       headers: { 'content-type': 'application/json' },
     });
+  }
+
+  /** fetcher is convenience fetch wrapper that includes authentication
+  * and other necessary headers **/
+  private async fetcher(url: string, init?: RequestInit) :Promise<any> {
+    const { token: idToken } = await this.identityApi.getCredentials();
+    const baseInit = {
+      headers: {
+        'Content-Type': 'application/json',
+        ...(idToken && { Authorization: `Bearer ${idToken}` }),
+        ...init?.headers
+      }
+    }
+    const newinit = Object.assign(baseInit, init);
+    const response = await fetch(url, newinit);
+    if (!response.ok) {
+      throw await ResponseError.fromResponse(response);
+    }
+    if (newinit.method === "DELETE") {
+        return response
+    }
+    return await response.json();
+  }
+  // temp duplicate solution for text consumers
+  private async fetcherText(url: string, init?: RequestInit) :Promise<any> {
+    const { token: idToken } = await this.identityApi.getCredentials();
+    const baseInit = {
+      headers: {
+        'Content-Type': 'application/json',
```

Comment:
```suggestion
        'Content-Type': 'text/plain',
```

Reply:
I have 3 remarks about this code:
1. The `Content-Type` should be overridable by the caller.
2. It cannot be `text/plain` as @batzionb suggests because for instance in [line 53](https://github.com/janus-idp/backstage-plugins/pull/1599/files#diff-e02456b10b75ecb00d8c7c09d87ab9480374f41e4734178ba89625aed2da1d20R53) you are sending a JSON payload.

I suggest to destructure the `init.headers` at the end of the object literal, this will allow you to define some defaults and allow the users to override them when necessary.

Reply:
I intentionally made auth headers and content type unchanable by the caller. At this stage I don't see
any other content type we need to support, and all calls should include identity information, if present. 
Unless I'm missing a use-case, why should we allow those behaviours if are not needed or wanted?

Reply:
If ["a GET request should contain no body"](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET#:~:text=Note%3A%20Sending%20body/payload%20in%20a%20GET%20request%20may%20cause%20some%20existing%20implementations%20to%20reject%20the%20request%20%E2%80%94%20while%20not%20prohibited%20by%20the%20specification%2C%20the%20semantics%20are%20undefined.%20It%20is%20better%20to%20just%20avoid%20sending%20payloads%20in%20GET%20requests.), then specifing a content-type in this kind of operation is redundant.